bin/build.sh

```diff
@@ -20,9 +21,15 @@ do
     SRC_NAME=$f
     OBJ_NAME=$(dirname ${f})/$(basename ${f} .lua).luac
     echo -e "Compiling file \e[1m${SRC_NAME}\e[21m..."
-    luac -s -o ${OBJ_NAME} ${SRC_NAME} || \
-    echo -e "\e[1m\e[39m[\e[31mBUILD FAILED\e[39m]\e[21m Compilation error in file \e[1m${SRC_NAME}\e[21m!" || \
-    exit 1
+    luac -s -o ${OBJ_NAME} ${SRC_NAME}
+    _fail=$?
+    if [[ $_fail -ne 0 ]]; then
+        FAIL=$((FAIL | _fail))
+        echo -e "\e[1m\e[39m[\e[31mBUILD FAILED\e[39m]\e[21m Compilation error in file ${SRC_NAME}\e[1m"
+    fi
 done
 
-echo -e "\e[1m\e[39m[\e[32mTEST SUCCESSFUL\e[39m]\e[21m All lua files built successfully!"
+if [[ $FAIL -eq 0 ]]; then
+    echo -e "\e[1m\e[39m[\e[32mTEST SUCCESSFUL\e[39m]\e[21m All lua files built successfully!"
+fi
+exit $FAIL
```

Comment:
Exiting with a status that is a logical or of all exit codes is a bit awkward, and makes it impossible to do something meaningful with the exit status. We should probably rather exit with the last failed compilation exit status.